pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -252,7 +253,7 @@
     private PulsarResources pulsarResources;
 
     private TransactionPendingAckStoreProvider transactionPendingAckStoreProvider;
-    private final ScheduledExecutorService transactionReplayExecutor;
+    private final ExecutorProvider internalExecutorService;
```

Comment:
```suggestion
    private final ExecutorProvider transactionExecutorProvider;
```

Reply:
And we can remove `numTransactionReplayThreadPoolSize` in the ServiceConfiguration.